File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/comment/comment/direct/Java/C++/s380356166_commented.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solver {
public:
    void solve(int testNumber) {
        int ans = 0;
        int K, S;
        cin >> K >> S;
        for (int x = 0; x <= K; x++) {
            for (int y = 0; y <= K; y++) {
                int z = S - x - y;
                if (z >= 0 && z <= K) {
                    ans++;
                }
            }
        }
        cout << ans << endl;
    }
};

int main() {
    Solver solver;
    solver.solve(1);
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/multiline/comment/direct/Java/Python/s380356166_commented.py
import sys

class Solver:
    def solve(self, testNumber, in_stream, out):
        ans = 0
        K = int(in_stream.readline())
        S = int(in_stream.readline())
        for x in range(K + 1):
            for y in range(K + 1):
                z = S - x - y
                if 0 <= z <= K:
                    ans += 1
        out.write(f"{ans}\n")

def main():
    in_stream = sys.stdin
    out = sys.stdout
    solver = Solver()
    solver.solve(1, in_stream, out)

if __name__ == "__main__":
    main()
# End of Code
